src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Completion/CompletionListMerger.cs

```diff
@@ -87,7 +89,7 @@ public static bool TrySplit(object? data, [NotNullWhen(true)] out IReadOnlyList<
             return false;
         }
 
-        var collector = new List<JsonElement>();
+        using var _ = ListPool<JsonElement>.GetPooledObject(out var collector);
```

Comment:
Nice catch <3 

Would it be better to use a pool and change the out to Immutable here? We can use `PooledArrayBuilder` and pass as ref because everything is sync, then assign the out after

Reply:
Yeah, that's probably better. It looks like the callers would be happy with an `ImmutableArray`.